02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -170,26 +170,285 @@ Here's what has been added to the above since then:
-    - Hard to keep documentation momentum
-    - Which libraries are 'core libraries'?
-- No clearly-defined mutually-held community-wide goal
-    - A desire to help exists, but no one is coordinating resources strategically
-- What exactly are FP's "Best Practices?"
-    - They are not well-defined, are assumed, or are habits
-    - Knowledge silos exist, so it's hard to extract that info
-- Slack-based discussions do not persist
+These are the reasons. Each wil be covered more in-depth following this:
+
+- PureScript is not _currently_ trying to be the next "mainstream language".
+- Only core contributors have write/deployment privileges to PureScript websites
+- Core contributors are spread thin
+- Breaking changes outdate documentation and kill documenters' motivation
+- There is no roadmap that coordinates efforts
+- Some libraries cannot publish their docs due to OOM error
+- It's difficult to pin down and define what are Functional Programming's "best practices"
+- Questions answered on Slack do not persist, so they get re-asked and re-answered
+
+This section concludes with a [narrative summary](#summary-of-Why-Docs-are-Lacking).
+
+### PureScript is not _currently_ trying to be the next "mainstream language"
+
+A few have wrongly assumed that PureScript is trying to replace [insert your favorite web language here]. That is not the case.
+
+PureScript has the following philosophy:
+> Making a language "successful" (e.g. popular, industry-standard, taught in universities' curriculums, etc.) is not a justification for paying for "costs" that ruin the language, such as limiting the language's expressiveness, making it less safe, less secure, etc.
+>
+> Making the language easier to use and learn by improving it is good and supported.
+> Making the language easier to use and learn by crippling, downgrading, or dumbing down the language is rejected. Users will either learn it properly or use something else.
+>
+> Plenty of languages chose to sacrifice safety, security, efficiency, etc. to gain the benefits of "success."
+>
+> PureScript will not be another such language. Rather, it will try to progress in "the right way," even when it's inconvenient, makes it harder to learn, etc.
+>
+> ~ Summary of [A Response that Explains the Motto: 'Avoid (Success at All Costs)'](https://news.ycombinator.com/item?id=12056169) (edits made: replaced 'Haskell' with 'PureScript'; removed "research-related concepts")
+
+This philosophy is summarized as "avoid (success at all costs)."
+
+The current focus (see later point) is not on making documentation great. Rather, it's on implementing the language features that haven't yet been implemented.
+
+However, even if great documentation was the focus, one will still have to learn difficult concepts before they can be productive. This fact will never change.
+
+Thus, PureScript might not be the language for you:
+- If you value 'success' over 'costs,' use a different language.
```

Comment:
I think simply summarizing the above explanation in the words "success" and "costs" makes this hard to read because it is rather difficult to separate meaning from the normal connotations or meanings of these words. So, I suggest the following word replacements.

success => language popularity
cost => language difficulty (or density?)

It's important for me to say that the idea intro and explanation above this, such as the quote, is quite good. I think it's very important reading for new adopters of the language.

Reply:
I think this is a good point. "success" and "cost" keep more with the motto, but most people don't understand that motto anyways. Using vernacular language is better than jargon.

Initially, I wanted to suggest a word other than "language popularity." However, I think that really does sum it up well. A "popular" language generally has these things:
- commercial IDE support
- good onboarding process
- curriculum in schools, so more people know about it already
- more developers use it, so more people write libraries for it, so the ecosystem usually already has working solutions/libraries to many programming problems.

It's a bit more nuanced than that, but popularity is a good way to sum it up.

"Difficulty" and "Density" also summarize it well. In FP, you can write a lot of powerful code in a few lines that take other languages much more. In this sense, it is "dense", but one has to understand a lot to be able to do that effectively. Thus, it's also "difficult" because one only becomes productive when they effectively integrate 8 different ideas together into a program: monads, app structure, pure functions, ADTs, etc. However, I think the 'difficulty' idea should add the idea of a learning curve:  'difficulty learning curve.' That's the real reasons why FP is difficult. Once one gets past that curve, it's not really 'difficult' anymore (I think?)

Reply:
I think it's worth adding to this document our definition of "popular" and examples of the things that popular languages have (the bullet point list you have here).

Regarding "density", I think this goes back to the idea that the separation between Pure FP/PureScript principles, idioms, and patterns are not well-defined and are often inter-mixed/not distinguished in existing teaching materials, which makes it difficult to map existing programming language knowledge into PureScript. So, it might be worth adding this definition/way of explaining "density" to this document here.

Reply:
So, I ended up using "Value power and expressivity over popularity" as a language motto. It states things in the positive (what we do value) rather than negative (what we don't value)

Reply:
It should suffice for now - I think most people would agree with that phrasing.